include/swift/AST/Types.h

```diff
@@ -3167,13 +3166,13 @@ class AnyFunctionType : public TypeBase {
   /// Create an AnyFunctionType.
   ///
   /// Subclasses are responsible for storing and retrieving the
-  /// ExtInfo::Uncommon value if one is present.
+  /// ExtInfo::ClangTypeInfo value if one is present.
   AnyFunctionType(TypeKind Kind, const ASTContext *CanTypeContext,
                   Type Output, RecursiveTypeProperties properties,
                   unsigned NumParams, ExtInfo Info)
   : TypeBase(Kind, CanTypeContext, properties), Output(Output) {
     Bits.AnyFunctionType.ExtInfoBits = Info.Bits;
-    Bits.AnyFunctionType.HasUncommonInfo = Info.getUncommonInfo().hasValue();
+    Bits.AnyFunctionType.HasClangTypeInfo = Info.getClangTypeInfo().hasValue();
```

Comment:
Hey @varungandhi-apple, looking at this I was wondering if maybe worth to `ExtInfo` to have an `hasClangTypeInfo()` API? Just a thought, maybe something like this could not be a relevant gain but could be an improvement over the  `getClangTypeInfo().hasValue()` calls

Reply:
I deliberately avoided that. IMO it doesn't make sense to implement `hasXYZ()`, `getXYZ()` for every type and optional field; that's precisely what `Optional` is for, so you can just return an `Optional` and have the caller use that. If it was the case that a large number of callers want to check only for existence, I would've been fine with adding a convenience method, but that's not the case here.